packages/@cdktf/provider-generator/lib/get/generator/resource-parser.ts

```diff
@@ -17,23 +17,27 @@ import {
   AttributeModel,
 } from "./models";
 
-const classNames: string[] = [];
-
-const uniqueClassName = (className: string): string => {
-  if (classNames.includes(className)) {
-    className = `${className}A`;
-  }
-  classNames.push(className);
-  return className;
-};
-
 const isReservedClassName = (className: string): boolean => {
   return ["string"].includes(className.toLowerCase());
 };
 
 class Parser {
   private structs = new Array<Struct>();
 
+  private classNames: string[] = [];
+
+  constructor(unique_classnames: string[]) {
+    this.classNames = unique_classnames;
```

Comment:
`unique_classnames` should be camel cased.

There is also a neat syntactic sugar one can use in cases like this: `constructor(private classNames: string[]){}`, see [the handbook](https://www.typescriptlang.org/docs/handbook/2/classes.html#parameter-properties) for a better explanation